### DISCUSSION TITLE
What is the correct type for routeName and params?

### DISCUSSION THREAD

**Discussion by m4tkv** [react-navigation/react-navigation#11265] [Q&A]
I am trying to add types for props route and params. 
I understand that this is a type intersection error, but how to get around this - no ideas yet

<img width="637" alt="image" src="https://user-images.githubusercontent.com/43599677/222908288-dac2f3b9-5154-4b90-8026-a8bb44ee9e3e.png">

<img width="950" alt="image" src="https://user-images.githubusercontent.com/43599677/222908308-8345a2a7-c189-4adb-9815-b1d823fe1541.png">

```
 "typescript": "4.8.4"
 "eslint": "^8.19.0"
 "@react-navigation/core": "^6.4.8",
 "@react-navigation/stack": "^6.3.16",
 "react": "18.2.0",
 "react-native": "0.71.3"
```

**Comment by BeeMargarida** [comment]
There's a section on the docs for type checking - https://reactnavigation.org/docs/typescript

So in this case you would have something like the code below in your navigation setup
```
type StackParams = {
  home: undefined;
  profile: {id: number};
};

const Stack = createStackNavigator<StackParams>();
```

And in some route, like profile, you would have something like:
```
type Props = StackScreenProps<StackParams, 'profile'>;

function ProfileScreen({navigation}: Props): JSX.Element {
   ....
}
```

I'm assuming App is some route in your app and not the place where you are initializing the navigation container. If not, those props don't make sense there.
